client/src/index.test.ts

```diff
@@ -1,31 +1,142 @@
-import { BaseClient, RestRepository, XataError, XataRecord } from './';
-
-const fetch = jest.fn();
-const client = new BaseClient(
-  {
-    fetch,
-    apiKey: '1234',
-    databaseURL: 'https://my-workspace-5df34do.staging.xatabase.co/db/xata:main'
-  },
-  {}
-);
+import { BaseClient, RestRepository, XataClientOptions, XataError, XataRecord } from './';
 
 interface User extends XataRecord {
   name: string;
 }
 
-const users = new RestRepository<User>(client, 'users');
+const buildClient = (options: Partial<XataClientOptions> = {}) => {
+  const {
+    apiKey = '1234',
+    databaseURL = 'https://my-workspace-5df34do.staging.xatabase.co/db/xata',
+    branch = 'main'
+  } = options;
+
+  const fetch = jest.fn();
+  const client = new BaseClient({ fetch, apiKey, databaseURL, branch }, {});
+  const users = new RestRepository<User>(client, 'users');
+
+  return { fetch, client, users };
+};
```

Comment:
Refactored the ``client``/``users`` a little so they're built with a helper instead of globals (allows for tests to customize them).